### PR TITLE
fix regex to not mix self closing tags

### LINKED ISSUE #1022: String.stripTags mixed some tags
Sometimes stripTags mixes some tags, for example `<b>` and `<br/>`

The following example shows the problem better:
http://jsfiddle.net/kallelat/Taa67/1/

Happens in MT 1.2+

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/359955-string-striptags-mixed-some-tags?utm_campaign=plugin&utm_content=tracker%2F22069&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F22069&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by SergioCrisostomo**
fixes #1022

### CODE REVIEW

File: Source/Types/String.Extras.js [hunk 1]
```diff
@@ -95,8 +95,8 @@ var walk = function(string, replacements){
 };
 
 var getRegexForTag = function(tag, contents){
-	tag = tag || '';
-	var regstr = contents ? "<" + tag + "(?!\\w)[^>]*>([\\s\\S]*?)<\/" + tag + "(?!\\w)>" : "<\/?" + tag + "([^>]+)?>";
+	tag = tag || (contents ? '': '\\w+');
```
Comment (arian) [inline Source/Types/String.Extras.js]: Space before the `:`.

**Comment by arian** [comment]
Could you elaborate the change?

**Comment by SergioCrisostomo** [comment]
@arian sure, should have done that in the first place, but here it comes:

This change only applies to cases where `contents` has boolean value `false`, and so the change is from

    "<\/?" + tag + "([^>]+)?>";

to

    "<\/?" + tag + "\/?>|<" + tag + "[\\s|\/][^>]*>"

and in cases where `tag` has boolean value `false`, `tag` will be `'\\w+'`.


As it was `str.stripTags("b")` [would also strip `<br/>` tags](http://jsfiddle.net/Taa67/2/) since the regex would be `<\/?b([^>]+)?>`.

**Comment by anutron** [comment]
:+1:

**Pull request merged**